Mage/src/main/java/mage/target/targetadjustment/DamagedPlayerControlsTargetAdjuster.java

```diff
@@ -0,0 +1,56 @@
+package mage.target.targetadjustment;
+
+import mage.abilities.Ability;
+import mage.filter.Filter;
+import mage.filter.predicate.card.OwnerIdPredicate;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.players.Player;
+import mage.target.Target;
+import mage.target.targetpointer.FirstTargetPointer;
+
+import java.util.UUID;
+
+/**
+ * @author notgreat
+ */
+public class DamagedPlayerControlsTargetAdjuster implements TargetAdjuster {
+    private Target blueprintTarget = null;
+    private boolean owner;
+    /**
+     * Use with {@link mage.abilities.common.DealsCombatDamageToAPlayerTriggeredAbility} with setTargetPointer enabled,
+     * or {@link mage.abilities.common.OneOrMoreDealDamageTriggeredAbility} with "SetTargetPointer.PLAYER" or similar.
+     * Adjusts the target to only target something the damaged player controls (or owns with alternative constructor)
+     * And then removes the effects' target pointer that the triggered ability set
+     */
+    public DamagedPlayerControlsTargetAdjuster() {
+        this(false);
+    }
+    public DamagedPlayerControlsTargetAdjuster(boolean owner) {
+        this.owner = owner;
+    }
+
+    @Override
+    public void adjustTargets(Ability ability, Game game) {
+        if (blueprintTarget == null) {
+            blueprintTarget = ability.getTargets().get(0).copy();
+        }
+        UUID opponentId = ability.getEffects().get(0).getTargetPointer().getFirst(game, ability);
+        Player opponent = game.getPlayer(opponentId);
+        ability.getTargets().clear();
+        ability.getAllEffects().setTargetPointer(new FirstTargetPointer());
+        if (opponent == null) {
+            return;
+        }
+        Target newTarget = blueprintTarget.copy();
+        Filter filter = newTarget.getFilter();
+        if (owner) {
+            filter.add(new OwnerIdPredicate(opponentId));
```

Comment:
Must add runtime check to make sure blueprint filter don't have controller/owner predicate. See example with `EquipAbility`. Maybe such check can be shareable for each target adjusters with blueprints like "makeSureFilterWithoutOwnerOrControllerPredicate":

![shot_240626_150906](https://github.com/magefree/mage/assets/8344157/95546eef-6dde-4fec-aac8-7e1f77d7e920)